csharp/autobuilder/Semmle.Autobuild.CSharp/CSharpAutobuilder.cs

```diff
@@ -114,6 +133,92 @@ public override BuildScript GetBuildScript()
             return attempt;
         }
 
+        protected override void AutobuildFailureDiagnostic()
+        {
+            // if `ScriptPath` is not null here, the `BuildCommandAuto` rule was
+            // run and found at least one script to execute
+            if (this.buildCommandAutoRule is not null &&
+                this.buildCommandAutoRule.ScriptPath is not null)
```

Comment:
With the other proposed (and now implemented) change to move all the rule fields into their own class, there is only the `ScriptPath` check here now, so I don't think that needs to be abstracted over -- especially since we use the `ScriptPath` in the diagnostic message as well.